### ISSUE TITLE
Unable to copy from ostree to docker-archive or oci-archive without compression

### ISSUE THREAD

**Issue by dcode** [containers/skopeo#704]
Related to #702, if I try to export from ostree storage to docker-archive, I get similar results as previously reported.

```
mkdir -p repo archives
ostree --repo=repo init
skopeo copy docker://docker.io/library/alpine:latest ostree:docker.io/library/alpine:latest@${PWD}/repo
skopeo copy ostree:docker.io/library/alpine:latest@${PWD}/repo docker-archive:archives/alpine_latest.tar
```

**Output**:
```
Getting image source signatures
Copying blob 1bfeebd65323 done
FATA[0000] Error writing blob: archive/tar: write too long
```

Similarly, if I allow uncompressed layers in an OCI archive, it also fails.
```
skopeo copy ostree:docker.io/library/alpine:latest@${PWD}/repo oci-archive:archives/alpine_latest.tar --dest-oci-accept-uncompressed-layers
```

**Output**:
```
Getting image source signatures
Copying blob 1bfeebd65323 done
FATA[0000] Error writing blob: Size mismatch when copying sha256:1bfeebd65323b8ddf5bd6a51cc7097b72788bc982e9ab3280d53d3c613adffa7, expected 2789742, got 5843968
```

It appears to be size calculation mismatch in both instances.

Note that copy to `oci-archive` with the default settings works just fine (and of course saves disk space):
```
skopeo copy ostree:docker.io/library/alpine:latest@${PWD}/repo oci-archive:archives/alpine_latest.tar
```

**Output**:
```
Getting image source signatures
Copying blob 1bfeebd65323 done
Copying config 4fa153a824 done
Writing manifest to image destination
Storing signatures
```

**Version:**
```
skopeo --version
skopeo version 0.1.40-dev commit: 1e044e42246c98ef7e118a07d5b4089f4957b8d7
```

I'm using the package I built in COPR here: https://copr.fedorainfracloud.org/coprs/g/rocknsm/testing/package/skopeo/

**Comment by mtrmac** [comment]
That looks pretty similar to #702.

Note that the fix for #702 exists in the containers/image repo, but it has not yet been vendored into the skopeo repo; AFAICS that COPR package is building the current skopeo version without the fix. Sorry for the confusion.

**Comment by mtrmac** [comment]
This works for me now, and was probably fixed in https://github.com/containers/skopeo/commit/a36d81c55c91de002bf2db10bb39c1bad84536ef , i.e. should be available in 0.1.40.

(Note that future Skopeo versions will not build with OSTree support by default, though it is, at least currently, still possible to explicitly enable it using a build tag.)
